tests/unit/modules/test_cron.py

```diff
@@ -1502,22 +1501,41 @@ def test_write_cron_lines_non_root_aix(self):
                 crontab_cmd, python_shell=False, runas="non-root"
             )
 
-    def test_write_cron_lines_non_root_solaris(self):
+    def test_write_cron_lines_non_root_aix(self):
         """
         Assert that _write_cron_lines() is called with the correct cron command and user
-        OS: Solaris. User: non-root. Expected to run with runas argument.
+        OS: AIX. User: non-root.
+        Expected to run with runas argument.
         """
         temp_path = "some_temp_path"
         crontab_cmd = "crontab {}".format(temp_path)
 
         with patch.dict(cron.__grains__, {"os_family": "AIX"}), patch.dict(
             cron.__salt__, {"cmd.run_all": MagicMock()}
+        ), patch("salt.utils.files.fpopen", mock_open()), patch.dict(
+            cron.__salt__, {"file.user_to_uid": MagicMock(return_value=1)}
         ), patch(
-            "salt.modules.cron._check_instance_uid_match",
-            new=MagicMock(return_value=False),
+            "salt.utils.files.mkstemp", MagicMock(return_value=temp_path)
         ), patch(
-            "salt.utils.files.fpopen", mock_open()
-        ), patch.dict(
+            "os.remove", MagicMock()
+        ):
+            cron._write_cron_lines("non-root", "test 123")
+            cron.__salt__["cmd.run_all"].assert_called_with(
+                crontab_cmd, python_shell=False, runas="non-root"
+            )
+
+    def test_write_cron_lines_non_root_solaris(self):
+        """
+        Assert that _write_cron_lines() is called with the correct cron command and user
+        OS: Solaris. User: non-root.
+        Expected to run with runas argument.
+        """
+        temp_path = "some_temp_path"
+        crontab_cmd = "crontab {}".format(temp_path)
+
+        with patch.dict(cron.__grains__, {"os_family": "AIX"}), patch.dict(
```

Comment:
Was this patch supposed to be Solaris instead of AIX?

Reply:
That's great catch! I haven't touch that code. I checked the mistake is in the current master branch.
I fixed it. Thanks!